src/System.Runtime.Extensions/src/System/OperatingSystem.cs

```diff
@@ -35,6 +36,43 @@ internal OperatingSystem(PlatformID platform, Version version, string servicePac
             _servicePack = servicePack;
         }
 
+        private OperatingSystem(SerializationInfo info, StreamingContext context)
+        {
+            SerializationInfoEnumerator enumerator = info.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                switch (enumerator.Name)
+                {
+                    case "_version":
```

Comment:
Can this be just:
```
_version = (Version)info.GetValue("_version", typeof(Version));
_platform = (PlatformID)info.GetValue("_platform", typeof(PlatformID));
_servicePack = info.GetString("_servicePack");
```

I do not think it make sense to replicate all cruft from the old desktop implementation.